src/components/base/ucc_base_iface.h

```diff
@@ -38,6 +38,10 @@ typedef struct ucc_base_lib_iface {
 
 typedef struct ucc_base_context_params {
     ucc_context_params_t params;
+    int                  estimated_num_eps;
+    int                  estimated_num_ppn;
```

Comment:
How to get values for this, from config? Is this maximum? What happens, if I violate it? Is this relevant beyond UCP tl?

Reply:
1. Various ways to get them:
- e.g., in xccl we compute that information (by performing 1 bootstrap allgather on a context)
- we can (and SHOULD) ask this info from the runtime, probably yes via context config

2. Yes, it is estimated max
3. It is perf optimization in UCP, nothing crashes if you don't pass it but perf difference is HUGE
4. Might be relevant for SHM TL as well (in hcoll we use that info to implement shmseg bcol efficiently)